google/auth/identity_pool.py

```diff
@@ -91,8 +90,12 @@ def __init__(
             quota_project_id=quota_project_id,
             scopes=scopes,
         )
-        if isinstance(credential_source, dict):
+        if not isinstance(credential_source, dict):
```

Comment:
The docstring says a `Mapping` is accepted, so please either check for `collections.Mapping` or update the docstring.

Reply:
Done.  I used collections.abc.Mapping, though, as I was getting warnings for using the regular collections library